Templates/templates/CSharp/Base/SharedCSharp.template.tt

```diff
@@ -334,5 +334,40 @@ public string GetMethodRequestPrimitiveReturnTypeString(string type)
     }
 }
 
+/// <summary>
+/// Used in EntityCollectionResponse.cs.tt, EntityCollectionWithReferencesResponse and  MethodCollectionResponse.cs.tt
+/// to get the function to initialized the NextPageRequest object in the response of the collection request
+/// and to copy the Additional data to the CollectionPage instance
+/// </summary>
+/// <param name="iBaseClientTypeName"> type of the IBaseClient to use in creating the method</param>
+/// <returns>the string representation of the method</returns>
+public string GetInitializeCollectionPropertiesMethod(string iBaseClientTypeName)
```

Comment:
could we have the collection responses derive from a base class or implement an interface with a default implementation instead of generating this all over the responses?

Reply:
That makes sense. I will create a PR to the core library to expose this function for use.

Reply:
@baywet 
When I was looking into this, I also saw that using your idea(as implemented in Java) of adding the "@odata.nextLink" as a serializable property would make the function created in the base class potentially unnecessary since the serializer would make the property available for us. Furthermore, null checking the property would also be unnecessary since the "InitializeNextPageRequest" already does this for us. We can therefore eliminate the generated method by directly passing the deserialized `NextLink` property to the "InitializeNextPageRequest".

Please check out the latest update and let me know what you think.

Reply:
thanks for taking in the changes, I think it looks much cleaner now. A few minor things left.